wg-node-lifecycle/charter.md

```diff
@@ -0,0 +1,202 @@
+
+- As a cluster admin I want to have a simple interface to initiate a node drain/maintenance without
+  any required manual interventions.
+- As a cluster admin, I want to be able to observe the node drain via the API and check on its
+  progress. I also want to be able to discover workloads that are blocking the node
+  drain.
+- As a cluster admin, I want to be able to perform arbitrary actions after the node drain is
+  complete, such as resetting GPU drivers, resetting NICs, performing software updates or shutting
+  down the machine.
+- As a cluster admin, I want to reduce the cost of doing maintenance on my hardware accelerators by
+  using control-plane APIs to help coordinate maintenance and drain a Node.
+- To support the new features, node maintenance, scheduler, descheduler, pod autoscaling, kubelet,
+  and other actors should use a new eviction API to gracefully remove pods. This would enable new
+  migration strategies that prefer to surge (upscale) pods first rather than downscale them. It
+  would also allow other users/components to monitor pods that are gracefully removed/terminated
+  and provide better behaviour in terms of de/scheduling, scaling and availability.
+- As an end user, I would like more alternatives to blue-green upgrades, especially with special
+  hardware accelerators; it's far too expensive. I would like to choose a strategy on how to
+  coordinate the node drain and the upgrade to achieve better cost-effectiveness.
+- As a cloud provider, I need to perform regular maintenance on the hardware in my fleet. Enhancing
+  Kubernetes to help CSPs safely remove hardware will reduce operational costs.
+- As an end user or admin, I would like to use a mixture of on-demand and temporary spot instances
+  in my clusters to reduce cloud expenditure. Having more reliable lifecycle and drain mechanisms
+  for nodes will improve cluster stability in scenarios where instances may be terminated by the
+  cloud provider due to cost-related thresholds.
+- As a user, I want to prevent any disruption to my pet or expensive workloads (VMs, ML with
+  accelerators) and either prevent termination for a long period of time or have a reliable
+  migration path. Features like `terminationGracePeriodSeconds` are not sufficient as the
+  termination/migration can take hours if not days.
+- As a user, I want my application to finish all network and storage operations before terminating a
+  pod. This includes closing pod connections, removing pods from endpoints, writing cached writes
+  to the underlying storage and completing storage cleanup routines.
+- As a cluster admin, I want a node to be declared as fully drained after all volumes are unmounted
+  from it.
+
+## Deliverables
+
+The WG will coordinate requirement gathering, design, implementation, and progressing through
+graduation stages.
+
+The group will help coordinate existing KEPs graduation as well as exploring new APIs and scenarios.
```

Comment:
Same comment as before, introduce acronym before using it. This helps newcomers better understand what the document is talking about.